pkg/controller/nodeipam/node_ipam_controller.go

```diff
@@ -96,13 +96,15 @@ func NewNodeIpamController(
 	// Cloud CIDR allocator does not rely on clusterCIDR or nodeCIDRMaskSize for allocation.
 	if allocatorType != ipam.CloudAllocatorType {
 		if len(clusterCIDRs) == 0 {
-			klog.Fatal("Controller: Must specify --cluster-cidr if --allocate-node-cidrs is set")
+			klog.Warning("Controller: Must specify --cluster-cidr if --allocate-node-cidrs is set")
+			panic("Controller: Must specify --cluster-cidr if --allocate-node-cidrs is set")
```

Comment:
I would expect to return an error here and below and make all callers handle returned errors correctly

Reply:
I tried in https://github.com/pacoxu/kubernetes/pull/1776/files.

Then no panic in `NewNodeIpamController`. 
-  It will be called by`startNodeIpamController` 

https://github.com/pacoxu/kubernetes/blob/67e12536f9ab516bf76e2402848d130952e2b79e/cmd/cloud-controller-manager/nodeipamcontroller.go#L57-L67

Reply:
if there's an error, I would expect it to be saved there and returned by the constructed InitFunc

Reply:
we discussed this some years ago, and we were concerned that there was something relying on this behavior, I mean, that the cloud-controller-manager crashed if the ipam wasn't working

Reply:
we should do as Jordan says https://github.com/kubernetes/kubernetes/pull/116174#issuecomment-1452105654

Reply:
I updated the PR to use error-return instead of panic.